### PR TITLE
Reorganize Ember Can

### PR DESCRIPTION
**Pull request by alexbeeken**
This pull request adds ability checks for submenu routes and reorganizes the ability namespace. This way abilities are more specific and flexible in application. 

instead of just calling `can see overview` for overview>activities route, you can now say `can view data in overview/activities`.

Expand abilities to check for permissions inside explore
Test for canViewData ability inside explore

Add Abilities for Submenu Items
Test For Submenu Item Visibility
Test For Submenu Route Permission

Fix Test Name Redundancy

Reorganize ember can abilities into folders
Add a Helper for common ability functions
Mock Ui service in navigation integration tests to close all drop downs
Add comments to ability files

Add tests for redirecting to unauthorized route with null auth-token
Add ability to set null auth-token in mockSessionService test helper
Redirect to unauthorized in application route

Set default authToken on mockSessionService to null instead of fake data
Rename properties variable to options in mock service test helpers

### CODE REVIEW

**Comment by alexbeeken** [comment]
This is ready to merge whenever.

File: app/components/nav-vertical/template.hbs [hunk 1]
```diff
@@ -83,21 +99,27 @@
...
-              </a>
-            </li>
-            <li class='nav-vertical-item'>
-              <a href='/{{project.site}}/image_sets'>
-                Image Sets
-              </a>
-            </li>
+            {{#if (can "view data in findings/grids")}}
+              <li class='nav-vertical-item'>
+                <a href='/{{project.site}}/activities/1/grid'>
```
Comment (zulaica) [inline app/components/nav-vertical/template.hbs]: Looks like this Rails route is hard-coded.

File: app/components/nav-vertical/template.hbs [hunk 2]
```diff
@@ -45,32 +51,42 @@
...
-              Word Tree
-            </a>
-          </li>
-          <li class='nav-vertical-item'>
-            <a href='/{{project.site}}/admin/explore/marker'>
-              Concept Canvas
-            </a>
-          </li>
-        </ul>
+          {{#if (can "view data in explore/responses")}}
```
Comment (zulaica) [inline app/components/nav-vertical/template.hbs]: There are sprinkles of double-quotes when they should be single quotes in the vertical nav templates.
